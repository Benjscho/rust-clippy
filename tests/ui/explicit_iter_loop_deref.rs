#![deny(clippy::explicit_iter_loop_deref)]
#![allow(
    clippy::linkedlist,
    clippy::similar_names,
    clippy::needless_borrow,
    clippy::deref_addrof,
    dead_code
)]

use core::slice;
use std::collections::*;

fn main() {
    let mut vec = vec![1, 2, 3, 4];

    let rmvec = &mut vec;
    for _ in rmvec.iter() {}
    for _ in rmvec.iter_mut() {}

    for _ in &vec {} // these are fine
    for _ in &mut vec {} // these are fine

    for _ in (&mut [1, 2, 3]).iter() {}
}
