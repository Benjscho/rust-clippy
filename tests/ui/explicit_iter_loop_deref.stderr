error: it is more concise to loop over references to containers instead of using explicit iteration methods
  --> $DIR/explicit_iter_loop_deref.rs:17:14
   |
LL |     for _ in rmvec.iter() {}
   |              ^^^^^^^^^^^^ help: to write this more concisely, try: `&*rmvec`
   |
note: the lint level is defined here
  --> $DIR/explicit_iter_loop_deref.rs:1:9
   |
LL | #![deny(clippy::explicit_iter_loop_deref)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: it is more concise to loop over references to containers instead of using explicit iteration methods
  --> $DIR/explicit_iter_loop_deref.rs:18:14
   |
LL |     for _ in rmvec.iter_mut() {}
   |              ^^^^^^^^^^^^^^^^ help: to write this more concisely, try: `&mut *rmvec`

error: it is more concise to loop over references to containers instead of using explicit iteration methods
  --> $DIR/explicit_iter_loop_deref.rs:23:14
   |
LL |     for _ in (&mut [1, 2, 3]).iter() {}
   |              ^^^^^^^^^^^^^^^^^^^^^^^ help: to write this more concisely, try: `&*(&mut [1, 2, 3])`

error: the method `iter` doesn't need a mutable reference
  --> $DIR/explicit_iter_loop_deref.rs:23:14
   |
LL |     for _ in (&mut [1, 2, 3]).iter() {}
   |              ^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::unnecessary-mut-passed` implied by `-D warnings`

error: aborting due to 4 previous errors

